galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/routes/PropertyQueryRoute.java

```diff
@@ -0,0 +1,135 @@
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import dev.galasa.framework.api.common.InternalServletException;
+import dev.galasa.framework.api.common.QueryParameters;
+import dev.galasa.framework.api.common.ResponseBuilder;
+import dev.galasa.framework.api.common.ServletError;
+import dev.galasa.framework.spi.ConfigurationPropertyStoreException;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.IFramework;
+
+import static dev.galasa.framework.api.common.ServletErrorMessage.*;
+
+public class PropertyQueryRoute extends CPSRoute{
+
+    private static final String path = "/cps/(.*)/properties([?]?|[^/])+$";
+
+    public PropertyQueryRoute(ResponseBuilder responseBuilder, IFramework framework) {
+        super(responseBuilder, path , framework);
+    }
+
+   
+    @Override
+    public HttpServletResponse handleRequest(String pathInfo, QueryParameters queryParams,HttpServletRequest req, HttpServletResponse response)
+            throws ServletException, IOException, FrameworkException {
+        String namespace = getNamespaceFromURL(pathInfo);
+        String properties = getNamespaceProperties(namespace, queryParams);
+        return getResponseBuilder().buildResponse(response, "application/json", properties, HttpServletResponse.SC_OK); 
+    }
+
+    private String getNamespaceProperties(String namespace, QueryParameters queryParams) throws FrameworkException{
+        String properties = "";
+         try {
+            nameValidator.assertNamespaceCharPatternIsValid(namespace);
+            if (hiddenNameSpaces.contains(namespace)) {
+            ServletError error = new ServletError(GAL5016_CPS_HIDDEN_NAMESPACE_ERROR, namespace);
+			throw new InternalServletException(error, HttpServletResponse.SC_NOT_FOUND);
+            }
+            String prefix = queryParams.getSingleString("prefix", null);
+            String suffix = queryParams.getSingleString("suffix", null);
+            properties = getProperties(namespace, prefix, suffix);
+        }catch (FrameworkException f){
+            if (f instanceof InternalServletException){
+                throw f;
+            }else{
+                ServletError error = new ServletError(GAL5017_INVALID_NAMESPACE_ERROR,namespace);  
+                throw new InternalServletException(error, HttpServletResponse.SC_NOT_FOUND);
+            }
+        }
+        return properties;
+    }
+    
+    
+    private String getProperties(String namespace, String prefix, String suffix) throws ConfigurationPropertyStoreException {
+        Map<String, String> properties = getAllProperties(namespace);
+       
+        if (prefix != null){
+            properties = filterPropertiesByPrefix(properties,prefix);
+        }
+        if (suffix != null){
+            properties = filterPropertiesBySuffix(properties,suffix);
+        }
+        
+        JsonArray propertyArray = new JsonArray();
```

Comment:
I would put the conversion of a properties map to json in a separate method.
Then that method only does one thing, and this method strings together things = 2 types of methods.